### PR TITLE
Ndabaga Opinion creation and json api

### PR DESCRIPTION
**Pull request by imsickofmaps**
This is a pure clone of articles

### CODE REVIEW

**Comment by imsickofmaps** [comment]
:+1: from me... @smn ?

File: ndabaga/api.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        return query
+
+    def alter_list_data_to_serialize(self, request, data_dict):
+        # Modifying the data to provide only what is needed in the right
+        # form by removing the extra meta variables and editing the
+        # dictionary
+        if isinstance(data_dict, dict):
+            if 'meta' in data_dict:
+                del(data_dict['meta'])
+            if data_dict['objects'] != []:
```
Comment (smn) [inline ndabaga/api.py]: an empty list `[]` resolves to `False`, a list with anything in it resolves to `True` so you can do the following here instead:

``` python
if data_dist['objects']:
   ....
```

which is a bit cleaner.

**Comment by smn** [comment]
Getting 3 test failures on this branch, waiting for Travis to see if it's a problem on my machine or not:

```
(ve)$ ./manage.py test
nosetests --verbosity 1
Creating test database for alias 'default'...
Raven is not configured (logging is disabled). Please see the documentation for more information.
..........EEE..
======================================================================
ERROR: test_no_ndabaga_last_week (ndabaga.tests.test_api.TestAPINdabagaNoNewNdabaga)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sdehaan/Documents/Repositories/go-nike-ghr/ndabaga/tests/test_api.py", line 40, in test_no_ndabaga_last_week
    'api_name': 'api'})
  File "/Users/sdehaan/Documents/Repositories/go-nike-ghr/ve/lib/python2.7/site-packages/django/core/urlresolvers.py", line 496, in reverse
    return iri_to_uri(resolver._reverse_with_prefix(view, prefix, *args, **kwargs))
  File "/Users/sdehaan/Documents/Repositories/go-nike-ghr/ve/lib/python2.7/site-packages/django/core/urlresolvers.py", line 416, in _reverse_with_prefix
    "arguments '%s' not found." % (lookup_view_s, args, kwargs))
NoReverseMatch: Reverse for 'api_dispatch_list' with arguments '()' and keyword arguments '{'api_name': 'api', 'resource_name': 'ndabaga'}' not found.

======================================================================
ERROR: test_ndabaga_not_published (ndabaga.tests.test_api.TestAPINdabagaNotPublished)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sdehaan/Documents/Repositories/go-nike-ghr/ndabaga/tests/test_api.py", line 61, in test_ndabaga_not_published
    'api_name': 'api'})
  File "/Users/sdehaan/Documents/Repositories/go-nike-ghr/ve/lib/python2.7/site-packages/django/core/urlresolvers.py", line 496, in reverse
    return iri_to_uri(resolver._reverse_with_prefix(view, prefix, *args, **kwargs))
  File "/Users/sdehaan/Documents/Repositories/go-nike-ghr/ve/lib/python2.7/site-packages/django/core/urlresolvers.py", line 416, in _reverse_with_prefix
    "arguments '%s' not found." % (lookup_view_s, args, kwargs))
NoReverseMatch: Reverse for 'api_dispatch_list' with arguments '()' and keyword arguments '{'api_name': 'api', 'resource_name': 'ndabaga'}' not found.

======================================================================
ERROR: test_api_works (ndabaga.tests.test_api.TestNdabagaAPIWorks)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/sdehaan/Documents/Repositories/go-nike-ghr/ndabaga/tests/test_api.py", line 16, in test_api_works
    'api_name': 'api'})
  File "/Users/sdehaan/Documents/Repositories/go-nike-ghr/ve/lib/python2.7/site-packages/django/core/urlresolvers.py", line 496, in reverse
    return iri_to_uri(resolver._reverse_with_prefix(view, prefix, *args, **kwargs))
  File "/Users/sdehaan/Documents/Repositories/go-nike-ghr/ve/lib/python2.7/site-packages/django/core/urlresolvers.py", line 416, in _reverse_with_prefix
    "arguments '%s' not found." % (lookup_view_s, args, kwargs))
NoReverseMatch: Reverse for 'api_dispatch_list' with arguments '()' and keyword arguments '{'api_name': 'api', 'resource_name': 'ndabaga'}' not found.

----------------------------------------------------------------------
Ran 15 tests in 0.520s

FAILED (errors=3)
Destroying test database for alias 'default'...
```

**Comment by gitumarkk** [comment]
@smn mine seems to work not sure what might be wrong

**Comment by smn** [comment]
@gitumarkk @imsickofmaps Travis reporting the same error: https://travis-ci.org/praekelt/go-nike-ghr/jobs/9003794

**Comment by gitumarkk** [comment]
@smn for some reason https://github.com/praekelt/go-nike-ghr/commit/4a303e3368cdebfe79333111af31231a4fc66c16#L7L20 has been deleted, will push it again.

**Comment by gitumarkk** [comment]
@smn that was very strange I have added it back, try now and see.

**Comment by smn** [comment]
@gitumarkk thanks! landing.

**Pull request merged**